docs/content/3.middleware/2.guides/3.extensions.md

```diff
@@ -117,3 +121,104 @@ curl '{SERVER_DOMAIN}/api/magento/getProductBySku`' \
   -H 'content-type: application/json' \
   -d '[{ "id": 1, "name": "John" }]'
 ```
+
+### Creating a configurable extension
+
+In more complex cases, you might want to create an extension that can be configured.
+
+Imagine that you're preparing an extension that would add cache control headers to responses based on the configuration.
+It might look like this:
+
+```ts
+const cacheControlConfig = {
+  searchProducts: "public, max-age=3600",
+  getProduct: "public, max-age=3600",
+};
+
+const cacheControlExtension = {
+  name: "caching-extension",
+  hooks(req, res) {
+    return {
+      afterCall({ response, callName }) {
+        if (req.method !== "GET") {
+          console.log("[CACHING] It's not a GET request, skipping caching");
+          return response;
+        }
+
+        if (res.getHeader("set-cookie")) {
+          console.log(
+            "[CACHING] Response containing Set-Cookie header, skipping caching"
+          );
+          return response;
+        }
+
+        if (callName in cacheControlConfig) {
+          res.set("Cache-Control", cacheControlConfig[callName]);
+        }
+
+        return response;
+      },
+    };
+  },
+};
+```
+
+The mechanism in here is universal and can be used in more than one integration. To avoid code duplication, you can prepare a closure that will create a new extension with a given configuration.
+
+```ts
+const createCacheControlExtension = (config: Record<string, string>) => {
+  return {
+    name: "caching-extension",
+    hooks(req, res) {
+      return {
+        afterCall({ response, callName }) {
+          if (req.method !== "GET") {
+            console.log("[CACHING] It's not a GET request, skipping caching");
+            return response;
+          }
+
+          if (res.getHeader("set-cookie")) {
+            console.log(
+              "[CACHING] Response containing Set-Cookie header, skipping caching"
+            );
+            return response;
+          }
+
+          if (callName in config) {
+            res.set("Cache-Control", config[callName]);
+          }
+
+          return response;
+        },
+      };
+    },
+  };
+};
+```
+
+Then, you can use it in your `middleware.config.ts` file:
+
+```ts
+export const integrations = {
+  sapcc: {
+    // ...
+    extensions: (extensions) => [
+      ...extensions,
+      createCacheControlExtension({
+        searchProducts: "public, max-age=3600",
+      }),
+    ],
+  },
+  contentful: {
+    // ...
+    extensions: (extensions) => [
+      ...extensions,
+      createCacheControlExtension({
+        getEntries: "public, max-age=3600",
+      }),
+    ],
+  },
+};
+```
+
+Using closures is a recommended way to create a configurable extension. It allows you to reuse the same logic in multiple integrations and keeps your code clean and maintainable.
```

Comment:
I think we can move this paragraph before the closure code snippet.

Reply:
@mattmaribojoc suggestion applied!